### PR TITLE
Unsettled amounts

### PR DESCRIPTION
**Pull request by afmhenry**

### CODE REVIEW

File: account-history/unsettled-amounts/demo.js [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+                "headers": {
+                    "Authorization": "Bearer " + document.getElementById("idBearerToken").value
+                }
+            }
+        ).then(function (response) {
+            if (response.ok) {
+                response.json().then(function (responseJson) {
+                    if (scope === "AmountTypes" && responseJson.Currencies.length > 0) {
+                        resetDropdownOptions("CurrencyAndAmountType")
+                        responseJson.Currencies.forEach(elem => {
```
Comment (basgroot) [inline account-history/unsettled-amounts/demo.js:120]: elem should be 'currency'

File: account-history/unsettled-amounts/demo.js [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    }
+
+    /**
+     * Populate dropdown with owed currencies and amount types
+     * @param {*} response 
+     */
+    function setDropdownForInstrument(response) {
+        var select = document.getElementById("idCbxCurrencyAndAmountType");
+        var currency = response.Currency
+        response["AmountTypes"].forEach(entry => {
```
Comment (basgroot) [inline account-history/unsettled-amounts/demo.js:48]: Preferred: response.AmountTypes.forEach

File: account-history/unsettled-amounts/demo.js [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+            var opt = document.createElement("option");
+            opt.value = currency + "-" + entry["AmountTypeId"]
```
Comment (basgroot) [inline account-history/unsettled-amounts/demo.js:50]: same here

File: account-history/unsettled-amounts/demo.js [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+        var select = document.getElementById("idCbx" + identifier)
+        var sel_len = select.length
+        for (var i = 0; i < sel_len; i++) {
+            select.remove(0)
+        }
+    }
+
+    /**
+     * Parse key information out of response and format for display
+     * @param {*} response
```
Comment (basgroot) [inline account-history/unsettled-amounts/demo.js:69]: Annotate real types (examples Object, string number, boolean)

File: account-history/unsettled-amounts/demo.js [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+        return "Endpoint: \n\t" + requestUrl + "\nParameters: \n\t" + params + "\n" + total + "\n" + dateRange + "\n" + details + "\n\n"
+
+    }
+    /**
+     * Get the unsettled amounts for a specific client, by currency or amount types
+     * @return {void}
+     */
+    function getUnsettledAmountsByCurrency(scope) {
+
+        let parameters = "?ClientKey=" + encodeURIComponent(demo.user.clientKey) + "&Scope=" + scope
```
Comment (basgroot) [inline account-history/unsettled-amounts/demo.js:104]: Either use 'var' consistently, or use 'let/const'. I prefer the latter, but it's a choice.

File: account-history/unsettled-amounts/demo.js [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+                demo.processError(response);
+            }
+        }).catch(function (error) {
+            console.error(error);
+        });
+    }
+    demo.setupEvents([
+        { "evt": "click", "elmId": "idBtnGetUnsettledAmountsByCurrency", "func": function () { getUnsettledAmountsByCurrency("Currencies") }, "funcsToDisplay": [getUnsettledAmountsByCurrency], "functionToRun": getUnsettledAmountsByCurrency },
+        { "evt": "click", "elmId": "idBtnGetUnsettledAmountsByAmountType", "func": function () { getUnsettledAmountsByCurrency("AmountTypes") }, "funcsToDisplay": [getUnsettledAmountsByCurrency], "functionToRun": getUnsettledAmountsByCurrency },
+        { "evt": "click", "elmId": "idBtnGetUnsettledAmountsByInstruments", "func": function () { getUnsettledAmountsByInstruments() }, "funcsToDisplay": [getUnsettledAmountsByInstruments], "functionToRun": getUnsettledAmountsByInstruments },
```
Comment (basgroot) [inline account-history/unsettled-amounts/demo.js:237]: I don't understand this. Will this be picked up by the boilerplate.js?

**Review by basgroot — CHANGES_REQUESTED** [review]

File: account-history/unsettled-amounts/demo.js (hunk 6 above)
Comment (afmhenry) [reply account-history/unsettled-amounts/demo.js:237]: you mean the "func" key? ? It is just a wrapper function, so I can call some endpoints with input params. Or do you mean something else?

Comment (afmhenry) [reply account-history/unsettled-amounts/demo.js]: Think i removed the confusing part--let me know

File: account-history/unsettled-amounts/demo.js (hunk 1 above)
Comment (basgroot) [reply account-history/unsettled-amounts/demo.js:122]: Why not:
responseJson.Currencies.forEach(currency => {

Comment (afmhenry) [reply account-history/unsettled-amounts/demo.js:122]: missed this apparently? thought i resolve it :)

Comment (afmhenry) [reply account-history/unsettled-amounts/demo.js]: fixed this part now

**Review by basgroot — APPROVED** [review]

**Pull request merged**